wireless_modules/battery_reader.py

```diff
@@ -0,0 +1,54 @@
+from machine import Pin, ADC
+from sensor_base import Sensor
+
+
+class BatteryReader(Sensor):
+    def __init__(self, pin_num=33, voltage_factor=None):
+        """
+        Initiate the battery reader algorithm
+        :param pin_num: The pin number on the ESP32 to read off the battery voltage from
+        :param voltage_factor: The factor to multiply the voltage read from the given pin to get the actual battery
+        voltage (Use the voltage divider formula to find this factor)
+        """
+        # Resistor values in the voltage divider design here:
+        # https://www.notion.so/Preliminary-Design-Report-2d0a06e271614ae886e7a2b3f88f93aa
+        self.R1 = 33.2
+        self.R2 = 100
+
+        # The factor to multiply the voltage at the battery pin with to get the battery voltage
+        self.voltage_factor = voltage_factor
+        if voltage_factor is None:
+            self.voltage_factor = (self.R1 + self.R2) / self.R1
```

Comment:
From my conversation with Brenny (@bdinh123), we're planning to make the same voltage divider (i.e same resistor values) for each of the WMs. Not sure if that's changed now but if not, then having a default `voltage_factor` would allow the three WM `main.py` scripts to not need to care about the circuit logic and design. And if the user needs to specify separate resistor values, they still can do that through passing their own `voltage_factor` argument.

Reply:
Fair point! In that case I think we just need a bit more flexibility. In my testing, the resistor values on their own don't give a perfect reading, I had to scale my voltages up a few percent in order to be accurate (probably something to do with the ADC). So we probably need something like the ability to scale the calculations from the resistors, rather than only the ability to override entirely.

Reply:
Do you think the scaling factor you used would be constant across all ESP32s? If so we can just have that inside the class. If not, we can let that be an argument to `__init__` - there's only 2 arguments at the moment so I think this approach should be fine but I feel like there's probably a better solution I can't think of right now.

Reply:
I'm not sure tbh... I think I'd rather have it as an argument though. It's the sort of thing we might want to change at short notice and this might make it easier!

Reply:
I think the best solution would be to have `voltage_factor` default to 1, and multiply the resistor-based calculation here by the voltage factor

Reply:
But that would mean the resistor values won't be easily modifiable. And a scenario where we decide to use different resistors for a given WM can occur, so maybe we should have both the resistor-based calculation and the scaling factor as an argument?

Reply:
Sorry, I didn't see this reply earlier somehow. I think that would be ideal! We want to avoid having to touch anything except `main.py` when running these.